tinyiiod.c

```diff
@@ -24,10 +24,13 @@ struct tinyiiod {
 	const struct tinyiiod_ops *ops;
 };
 
+static char *iiod_buf;
```

Comment:
Wouldn't be better to have the iiod_buf as a member of struct tinyiiod?

Reply:
The good side is that it is suitable for multiple instances of "struct tinyiiod".
The bad side is that whoever instantiates "struct tinyiiod" would have access to iiod_buf. The user should only use tinyiiod trough "const struct tinyiiod_ops *ops" interface.

Reply:
To limit access to the buffer you can use a mutex lock to limit availability of the resource to other instances.

Reply:
This following cases apply for "iiod_buf" when declared as a single global variable and we have more than one instance of "struct tinyiiod" (one for serial and one for network):

- Even if we have multiple instances, a mutex wont be necessary, since the code is single threaded, in our case.
- If libtinyiiod is used in a multi thread application, then we might have a problem and a mutex should be used. The mutex implementation should be outside libtinyiiod, and should be accessible trough "const struct tinyiiod_ops *ops;" element of "tinyiiod".